### PR TITLE
Remove dependency on a private PETSc header file

### LINKED ISSUE #7312: snesimpl
### Description of the enhancement or error report

I can't compile PetscSupport.C   I need the following include line:

#include <petsc-private/snesimpl.h>

instead of

#include <petsc/private/snesimpl.h>

I have petsc/3.4.4

Tagging @fdkong

### PR DESCRIPTION
**Pull request by fdkong**
The location of PETSc private heard files sometime changes from version to version, and we so remove dependency on a private PETSc header file, snesimpl.h.

Closes #7312

### CODE REVIEW

File: framework/src/utils/PetscSupport.C [hunk 1]
```diff
@@ -150,18 +150,18 @@ petscSetupDM (NonlinearSystem & nl) {
   PetscNonlinearSolver<Number> *petsc_solver = dynamic_cast<PetscNonlinearSolver<Number> *>(nl.sys().nonlinear_solver.get());
   SNES snes = petsc_solver->snes();
   // if there exists a DMMoose object, not to recreate a new one
-  if (snes->dm)
+  ierr = SNESGetDM(snes,&dm);
```
Comment (permcody) [inline framework/src/utils/PetscSupport.C]: space between function arguments

File: framework/src/utils/PetscSupport.C [hunk 2]
```diff
@@ -150,18 +150,18 @@ petscSetupDM (NonlinearSystem & nl) {
...
+  CHKERRABORT(nl.comm().get(),ierr);
```
Comment (permcody) [inline framework/src/utils/PetscSupport.C]: space between function arguments

**Comment by permcody** [comment]
I discussed the formatting issues with @fdkong. He wants to fix all of PetscSupport (which has several formatting issues). I'll merge this as is and let him put up a second PR with all the formatting fixes.

**Comment by WilkAndy** [comment]
Thanks for fixing the PETSc stuff, @fdkong .  It works for me now,

a

**Comment by jwpeterson** [comment]
@fdkong while you're at it, maybe you can eventually fix the `#include<petsc/private/dmimpl.h>` in PetscDMMoose.C too.

**Comment by fdkong** [comment]
@jwpeterson, it is not easy to remove this one because PETSc does not have a good way (via a public interface) to pass member function pointers to DM. We hence have to assign pointers directly.

**Pull request merged**